### ISSUE TITLE
More robust attribution (via GitHub login, versus display name?)

### ISSUE THREAD

**Issue by jimallman** [OpenTreeOfLife/opentree#610]
If a curator changes the display name in their GitHub account, their contributions become fragmented in oti (and Nexson). For example, note the different results in the curation home page:

https://devtree.opentreeoflife.org/curator/?match=jonathan+rees

https://devtree.opentreeoflife.org/curator/?match=jonathan+a+rees

Consider storing GitHub userids in tandem with the `ot:curatorName` list, or as part of it? This will still fail if someone changes their GitHub login, but so (I'm guessing) will every other form of attribution on GitHub, so at least they'll be expecting it. 

_N.B. We store the curators' display names instead of userids because that's generally what we want to show. Otherwise we'd need to translate from userid to display name in many places._

**Comment by kcranston** [comment]
Hmmm... I suppose we could do a periodic cleanup of curatorName (matching to current GitHub display names). Sounds like we might want a `ot:curatorUserId` field in addition to `ot:curatorName` and perhaps an option on the curator page to update name on curated studies?

I am less concerned about people changing their GitHub userids. GitHub is pretty clear that this creates havoc.

**Comment by jimallman** [comment]
Yes, I've considered adding a second list as you suggested, and keeping them in tandem:
```json
"ot:curatorName": ["Karen Cranston", "Jim Allman"],
"ot:curatorUserId": ["kcranston", "jimallman"],
```
This sort of thing makes me itch, since it's easy to imagine these getting out of synch. But there's only one place in my code that manipulates curator information, so it seems manageable.

Alternately, we might consolidate the information like so:
```json
"ot:curators": [
    {"userid": "kcranston" , "name": "Karen Cranston"}, 
    {"userid": "jimallman" , "name": "Jim Allman"}
], 
```
I prefer this solution, but it would break more code in the short term. I think either of these changes would prompt a sweep of the existing phylesystem data.

**Comment by jimallman** [comment]
After further discussion (all-hands meeting, Sept 2019), we've settled on an approach that balances backward compatibility and long-term utility (and consistency with our other JSON docs):

  - build a table of DISPLAY_NAME to GITHUB_USERID for all curators
    - gather ALL distinct values found in '^ot:curatorName'
      - use Python on local (production 'api') repo
      - make local modifications, then push/update all at once (should force re-index)
    - follow name changes, etc.; conform to latest/current name on Github?
    - list the proper/current Github userid for each person
  - refactor Nexson to handle this
    - ADD '^ot:curators' property with a list [patterned on our collections JSON](https://github.com/OpenTreeOfLife/collections-1/blob/657a6c62c2c07013815e0a1d22ee05db6cc67b39/collections-by-owner/opentreeoflife/default.json), e.g.
      ```json
      "^ot:curators": [
        {
          "login": "DMaddison", 
          "name": "David Maddison"
        }, 
        {
          "login": "cdspecht", 
          "name": "cdspecht (None)"
        }, 
        {
          "login": "mtholder", 
          "name": "Mark T. Holder"
        }
      ]
      ```
    - MOVE existing "^ot:curatorName" value(s) to "^ot:curators"[name]
      ```js
      "^ot:curatorName": "Jiabin Deng", 
      ```
      OR perhaps:
      ```js
      "^ot:curatorName": [
        "LukeKlicka", 
        "Mark T. Holder"
      ],
      ```
    - REPLACE existing "^ot:curatorName" with
      ```js
      "^ot:curatorName": "MOVED TO '^ot:curators', see {Nexson-doc-URL}", 
      ```
      OR
      ```
      "^ot:curatorName": [
        "MOVED TO '^ot:curators' property", 
        "For details, see {Nexson-doc-URL}"
      ],
      ```
    - CORRECT new "^ot:curators" to include name AND Github userid
      (based on a small table prepared above)
      - add userid
      - conform to latest/best display name for this id on Github
    - UPDATE some kind of Nexson version identifier? I don't see one yet...

I believe @mtholder is currently holding the baton, using [this script from @jimallman](https://gist.github.com/jimallman/3a0196299b846958793aa8ba98e3bf69) to gather all current curator information from [the production phylesystem](https://github.com/OpenTreeOfLife/phylesystem-1).
